### PR TITLE
Leverage pre-commit framework

### PR DESCRIPTION
**Pull request by inverse**
- Leverage the pre-commit framework - more rules can be added but this should suffice for now
- Split up CI to lint as sep flow (speed + clarity)

### CODE REVIEW

**Comment by ZENALC** [comment]
Very nice PR again.

**Comment by inverse** [comment]
Could also be improved to leverage [caching](https://github.com/actions/cache). But I'll save that for another PR :)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
 # Contributing
 
 I would love to see people collaborate with me on this project of making the most advanced cryptocurrency trading bot on the planet. If you'd like to help out, you can always submit pull requests. Or if you'd like to see features, you can always feel free to do so through Github's issues tab.
+
+## Pre-commit
+
+We leverage the [pre-commit](https://pre-commit.com/) framework.
+
+Install git hooks with `pre-commit install` or run the checks manually `pre-commit run`. Passing `-a` to run over all files.
```
Comment (inverse) [inline CONTRIBUTING.md:9]: This part is purely optional as it runs on CI :)

**Review by ZENALC — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (ZENALC) [reply CONTRIBUTING.md:9]: It's still a good feature to have :)

Comment (ZENALC) [reply CONTRIBUTING.md:9]: It's still a great thing to have :)

**Comment by ZENALC** [comment]
@inverse Caching sounds awesome!

**Pull request merged**